pkg/controller/manager.go

```diff
@@ -68,12 +65,16 @@ func (m *Manager) UpdateController(name string, params ControllerParams) *Contro
 		params.StopFunc = NoopFunc
 	}
 
+	m.mutex.Lock()
+
```

Comment:
`m.controllers` is always set in NewManager. I like knowing that this function will do the right thing but it might be simpler if we just remove the `if m.controllers...` part and avoid the lock altogether.

Reply:
We'll have to take the lock anyway unless we make it the problem of the caller to maintain owning the controller pointer.